exercises/isbn-verifier/tests/isbn-verifier.rs

```diff
@@ -75,6 +75,18 @@ fn test_invalid_isbn_too_long() {
     assert!(!is_valid_isbn("3-598-21507-XX"));
 }
 
+#[test]
+#[ignore]
+fn test_valid_digits_invalid_length() {
+    assert!(!is_valid_isbn("35982150881"));
+}
+
+#[test]
+#[ignore]
+fn test_special_characters() {
+    assert!(!is_valid_isbn("!@#%!@"));
```

Comment:
What implementation would have accepted this in the first place?

Reply:
```rust
isbn.chars()
        .filter(|ch| (ch.is_digit(10)) || (*ch == 'X' && isbn.find(*ch).unwrap() == isbn.len() - 1))
        .map(|ch| {
            if ch == 'X' {
                10
            } else {
                ch.to_digit(10).unwrap()
            }
        })
        .enumerate()
        .fold(0, |acc, (i, num)| acc + (10 - (i as u32)) * num) % 11 == 0
```

Reply:
Huh. You're right. I revoke my objection.